src/main/kotlin/com/exactpro/th2/common/schema/configuration/ConfigurationManager.kt

```diff
@@ -73,23 +68,7 @@ class ConfigurationManager(private val configurationPath: Map<Class<*>, Path>) {
             LOGGER.error { "Mapping does not contain module factory for $moduleClass" }
             "Mapping does not contain module factory for $moduleClass"
         }
-        val module: T = moduleFactory.loadModule(configurationProvider, moduleClass) as T
-        checkNotNull(module) {
-            LOGGER.error { "Returned null instead of instance of class $moduleClass" }
-            "Returned null instead of instance of class $moduleClass"
-        }
-        return module
-    }
-
-    fun <T : Configuration> getConfigurationWithConfigurationProvider(
-        configClass: Class<T>,
-        configurationProvider: ConfigurationProvider
-    ): T {
-        return cache.computeIfAbsent(configClass) {
-            checkNotNull(configurationFactoryMapping[configClass]) {
-            "Unknown class $configClass"
-        }.loadConfiguration(configurationProvider, configClass)
-        } as T
+        return moduleFactory.create(configurationProvider) as T
```

Comment:
Why do we always create a new module instead of caching the first result and returning it on all other calls